spec/generator.json

```diff
@@ -31,10 +31,19 @@
             ]
         },
 
+        "file": {
+            "description": "Represents a source file",
+            "fields": [
+                { "name": "name", "type": "string", "description": "The recommended name for the file, including the path to the file, where appropriate.", "example": "com/myorg/myapplication/Foo.java" },
```

Comment:
OK - have added a 'dir' field that is optional. apidoc-generator/master is
fully up to date and ready for release (will plan to release tonight). Note
I've added default file names and directories based on languages (see
https://github.com/mbryzek/apidoc-generator/blob/master/generator/test/lib/ServiceFileNamesSpec.scala
)

apidoc changes should be good to go to:
https://github.com/mbryzek/apidoc/pull/351

If you can take a quick look, can deploy tomorrow.

I've verified that the changes are in fact backwards compatible (using
apidoc-cli). Once released, I'll plan to update apidoc-cli to use the
directory and file names, removing them from the .apidoc config file (not
sure if you use apidoc-cli or not).

And then we should be clear to get the java client merged in.

Mike



On Tue, Jun 16, 2015 at 6:21 PM, John Kenny <notifications@github.com>
wrote:

> In spec/generator.json
> <https://github.com/mbryzek/apidoc/pull/351#discussion_r32578127>:
>
> > @@ -31,10 +31,19 @@
> >              ]
> >          },
> >
> > +        "file": {
> > +            "description": "Represents a source file",
> > +            "fields": [
> > +                { "name": "name", "type": "string", "description": "The recommended name for the file, including the path to the file, where appropriate.", "example": "com/myorg/myapplication/Foo.java" },
>
> I was thinking of splitting the 'name' field into 'name' and an optional
> 'path' field. I think this is more self-documenting and easier to consume.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mbryzek/apidoc/pull/351/files#r32578127>.
>